classes/03class/aws/README.md

```diff
@@ -0,0 +1,184 @@
+
+- [Introduction](#introduction)
+- [PET vs CATTLE](#pet-vs-cattle)
+- [AWS EC2 Auto Scaling Group (ASG)](#aws-ec2-auto-scaling-group-asg)
+  - [How it works](#how-it-works)
+  - [Automating EC2 instances](#automating-ec2-instances)
+- [AWS Load Balancer](#aws-load-balancer)
+- [Route53](#route-53)
+- [References](#references)
+
+## Introduction
+
+At this stage of the DevOps Academy we will start connecting the dots and understand why we have been learning this many different subjects.
+
+The project #1 kickoff is coming soon and will use most of the concepts that we have seen so far.
+
+This article will cover why you should be chosing CATTLE over PETS and how to implement.
+
+
+## PET vs CATTLE
+
+The analogy comes from how do you treat your servers. In the old days, you would even give a name to your server like, Thunder/Flash or Tiger/Monkey if you prefer animal names. Very much like a PET where you need to individually take care of it, install programs, patch things, etc. These days with automation, we now treat servers as cattle, where if one server dies, we just replace it.
+
+The idea is to avoid pet server as manual configuration tend to generate more maintenance.
+
+Pet servers have issues with scaling up and down. For example a server with 8GB of RAM memory hosting an e-commerce, if you are getting more clients than what you expect, you will have to shutdown the computer, install more RAM and start again.
+
+
+## AWS EC2 Auto Scaling Group (ASG)
+
+The service can be found at Services > Compute > EC2 > Auto Scaling > Create Auto Scaling group
+
+EC2 Auto Scaling ensures that you always have the amount of instances needed to handle the load of your application. Based on healthchecks and resources consumption logs, the asg will create new instances or shutdown the ones that are iddle.
+
+You could probably have been asking yourself: if I have two servers running my website, how my clients will access both of them?
+
+This is where [Load Balancers](#aws-load-balancer) will help balancing requests through your array of servers created by an ASG. [Load Balancers](#aws-load-balancer) will be introduced later in this article.
+
+ASG are described in 3 key components:
+
+- Groups
+  - a group treats an array of server as one logical resource. When creating a group you have to specify its minimum, maximum and desired number of EC2 instances.
+    - minimum: have at least this minimum of EC2 instances online
+    - maximum: have this maximum of EC2 instances online
+    - desired: try to keep this desired number
+
+![asg-concepts](/classes/03class/aws/assets/asg-concepts.png)
+
+- Configuration templates
+  - specify instance image ID (AMI ID), instance type, ssh key, security group and storage options
+
+- Scaling options
+  - scale up and down the number of instances based on set of defined rules
+
+There are no costs of creating an ASG by itself. The cost will be related with the number and type of EC2 instances that your group create.
```

Comment:
There no costs, change the second sentence to make it clear that costs are due to EC2 instances